kubeadm GA blog post

luxas opened the pull request:
/cc @fabriziopandini @neolit123 @roberthbailey 
/assign @kbarnard10 @lucperkins @timothysc

You can see a preview of the post in progress here:

https://deploy-preview-11462--kubernetes-io-master-staging.netlify.com/blog/2018/12/04/production-ready-kubernetes-cluster-creation-with-kubeadm/

dims commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,91 @@
+
+[kubeadm](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/) is a tool that enables Kubernetes administrators to quickly and easily bootstrap minimum viable clusters that are fully compliant with [Certified Kubernetes](https://github.com/cncf/k8s-conformance/blob/master/terms-conditions/Certified_Kubernetes_Terms.md) guidelines. It's been under active development by [SIG Cluste ...
+
+This GA release is an important event in the progression of the Kubernetes ecosystem, bringing stability to an area where stability is paramount.
```
`This GA release` -> `This GA release of kubeadm`

dims commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,91 @@
+
+This GA release is an important event in the progression of the Kubernetes ecosystem, bringing stability to an area where stability is paramount.
+
+The goal of kubeadm is to provide a foundational implementation for bootstrapping common cluster setup tasks and for cluster administration. kubeadm ships with best-practice defaults but can be customized to support other ecosystem requirements or vendor-specific approaches. kubeadm is designed to be easy to integrate into larger systems.
```
`but can be` -> `and can also be` (keep it positive)

dims replied:
`larger systems` -> `larger deployment systems/tools` (may make it clearer?)

dims commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,91 @@
+
+Infrastructure provisioning, for example, is left to other SIG Cluster Lifecycle projects, such as the [Cluster API](https://github.com/kubernetes-sigs/cluster-api). Instead, kubeadm covers only the common denominator in every Kubernetes cluster: the [control plane](https://kubernetes.io/docs/concepts/#kubernetes-control-plane). The user may install their preferred networking solution and other a ...
+
+The core of the kubeadm interface is quite simple: new control plane nodes are created by running `kubeadm init` and worker nodes are joined to the control plane by running `kubeadm join`. Also included are utilities for managing already bootstrapped clusters, such as control plane upgrades and token and certificate renewal.
```
we should move this to the first para (line 17). Otherwise the bulk of this sections seems to be about what kubeadm will not do!

dims commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,91 @@
+ * **Stable underlying implementation** — kubeadm now creates a new Kubernetes cluster using methods that shouldn't change any time soon. The control plane, for example, is run as a set of static Pods, bootstrap tokens are used for the **kubeadm join** flow, and [ComponentConfig](https://github.com/kubernetes/community/blob/master/keps/sig-cluster-lifecycle/0014-20180707-componentconfig-api-types ...
+ * **Configuration file schema** — With the new  **v1beta1** API version, you can now tune almost every part of the cluster declaratively and thus build a "GitOps" flow around kubeadm clusters. In future versions, we plan to graduate the API to version **v1** with minimal changes (and perhaps none).
+ * **The "toolbox" interface of kubeadm** — Also known as **phases**. If you don't want to consume the all of the **kubeadm init** tasks performed, you can perform fine-grained, generic actions with the **kubeadm init phase** command (for example, generating certificates or control plane Static Pod manifests).
+ * **Upgrades between minor versions** — The **kubeadm upgrade** command is now fully GA. It handles control plane upgrades for you, which includes upgrades to etcd, the API Server, the Controller Manager, and the Scheduler. You can seamlessly upgrade your cluster between minor or patch versions (e.g. v1.12.2 -> v1.13.1 or v1.13.1 -> v1.13.3)
```
do we want to say anything about `downgrades`?

timothysc replied:
Nope.

dims commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,91 @@
+ * **Configuration file schema** — With the new  **v1beta1** API version, you can now tune almost every part of the cluster declaratively and thus build a "GitOps" flow around kubeadm clusters. In future versions, we plan to graduate the API to version **v1** with minimal changes (and perhaps none).
+ * **The "toolbox" interface of kubeadm** — Also known as **phases**. If you don't want to consume the all of the **kubeadm init** tasks performed, you can perform fine-grained, generic actions with the **kubeadm init phase** command (for example, generating certificates or control plane Static Pod manifests).
+ * **Upgrades between minor versions** — The **kubeadm upgrade** command is now fully GA. It handles control plane upgrades for you, which includes upgrades to etcd, the API Server, the Controller Manager, and the Scheduler. You can seamlessly upgrade your cluster between minor or patch versions (e.g. v1.12.2 -> v1.13.1 or v1.13.1 -> v1.13.3)
+ * **etcd setup** — etcd is now set up in a way that both is secure by default, with TLS communication everywhere, and allows for expanding to a highly available cluster when needed
```
Do we want to say something about HA?

timothysc replied:
Not yet, it's still experimental.

dims commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,91 @@
+- You're a **new user** who wants to take Kubernetes for a spin. kubeadm is the fastest way to get up and running on [Linux machines](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/). If you're using [Minikube](https://github.com/kubernetes/minikube) on a Mac or Windows workstation, you're actually already running kubeadm inside the Minikube VM!
+- You're a **system administrator** responsible for setting up Kubernetes on bare metal machines and you want to quickly create Kubernetes clusters that are secure and in conformance with best practices and also configurable.
+- You're a **cloud provider** who wants to add a Kubernetes offering to your suite of cloud services. kubeadm is the go-to tool for creating clusters at a low level.
+- You're an **organization with special needs**. Existing public cloud offerings like [Amazon EKS](https://aws.amazon.com/eks/) and [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine/) won't cut it for you; you need customized Kubernetes clusters tailored to your hardware, security, policy, and other needs.
```
This point is counter to the others on the list. (kubeadm is not suitable)

nikopen replied:
why not? kubeadm is granular enough to be integrated in a 'custom installer' that calls on other APIs, like openstack, vmware, public clouds... et cetera

timothysc replied:
nak, the flow makes sense and is correct on re-read. 

Do you have a proposed alternative phrasing?

lucperkins commented:
@luxas I hijacked your branch and pushed some changes directly to the PR. Hope that's okay! Mostly I added proper formatting for the blog (page metadata), fixed the CSS for tables in blog posts, and addresses some issues pointed out by commenters.

timothysc commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,97 @@
+
+### The scope of kubeadm
+
+kubeadm is focused on bootstrapping Kubernetes clusters on existing infrastructure and performing an essential set of maintenance tasks. The core of the kubeadm interface is quite simple: new control plane nodes are created by running `kubeadm init` and worker nodes are joined to the control plane by running `kubeadm join`. Also included are utilities for managing already bootstrapped clusters, s ...
```
might want to link the subsections of the docs for `init` and `join`.

lucperkins replied:
Good call. I added links to docs all of the commands listed in the post.

fabriziopandini commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,97 @@
+
+This GA release of kubeadm is an important event in the progression of the Kubernetes ecosystem, bringing stability to an area where stability is paramount.
+
+The goal of kubeadm is to provide a foundational implementation for bootstrapping common cluster setup tasks and for cluster administration. kubeadm ships with best-practice defaults but can also be customized to support other ecosystem requirements or vendor-specific approaches. kubeadm is designed to be easy to integrate into larger deployment systems and tools.
```
"for bootstrapping common cluster setup tasks and for cluster administration." --> what about simplifying in "for cluster setup and for cluster administration."

fabriziopandini commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,97 @@
+
+### kubeadmn survey
+
+SIG Cluster Lifecycle has put together a survey for kubeadm. It's available at [https://bit.ly/2FPfRiZ](https://bit.ly/2FPfRiZ). The purpose of the survey is to collect community feedback about kubeadm for the sake of future improvement.
```
What about trying to do better marketing for the survey: e.g.
"Although kubeadm is GA, the SIG Cluster Lifecycle commitment to improving the user experience in managing Kubernetes clusters will continue. We are launching a survey is to collect community feedback about kubeadm for the sake of future improvement."

benmoss replied:
Also a typo in the header here: `kubeadmn`

luxas commented:
> @luxas I hijacked your branch and pushed some changes directly to the PR. Hope that's okay! Mostly I added proper formatting for the blog (page metadata), fixed the CSS for tables in blog posts, and addresses some issues pointed out by commenters.

Thanks super okay :100:, thank you for doing it! Feel free to continue copyediting as you see fit until we merge this tomorrow (4th of Dec, as decided with @kbarnard10 earlier)

kbhawkey commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -14,7 +14,7 @@ The goal of kubeadm is to provide a foundational implementation for bootstrappin
 ### The scope of kubeadm
 
-kubeadm is focused on bootstrapping Kubernetes clusters on existing infrastructure and performing an essential set of maintenance tasks. The core of the kubeadm interface is quite simple: new control plane nodes are created by running `kubeadm init` and worker nodes are joined to the control plane by running `kubeadm join`. Also included are utilities for managing already bootstrapped clusters, s ...
+kubeadm is focused on bootstrapping Kubernetes clusters on existing infrastructure and performing an essential set of maintenance tasks. The core of the kubeadm interface is quite simple: new control plane nodes are created by running [`kubeadm init`](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/) and worker nodes are joined to the control plane by running [`kubeadm join` ...
```
nit: capitalize Kubeadm when starting a sentence?

lucperkins replied:
This is not currently done in the kubeadm documentation (for example [here](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/)). I'm following the existing usage but am 100% happy to defer to you on that.

neolit123 replied:
lowercase kubeadm at the start of a sentence is fine.

kbhawkey commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -32,10 +32,10 @@ General Availability means different things for different projects. For kubeadm,
- * **The "toolbox" interface of kubeadm** --- Also known as **phases**. If you don't want to consume the all of the `kubeadm init` tasks performed, you can perform fine-grained, generic actions with the `kubeadm init phase` command (for example generating certificates or control plane [Static Pod](https://kubernetes.io/docs/tasks/administer-cluster/static-pod/) manifests).
- * **Upgrades between minor versions** --- The `kubeadm upgrade` command is now fully GA. It handles control plane upgrades for you, which includes upgrades to [etcd](https://etcd.io), the [API Server](https://kubernetes.io/docs/reference/using-api/api-overview/), the [Controller Manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/), and the [Schedul ...
+ * **The "toolbox" interface of kubeadm** --- Also known as **phases**. If you don't want to consume the all of the [`kubeadm init`](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/) tasks performed, you can perform fine-grained, generic actions with the `kubeadm init phase` command (for example generating certificates or control plane [Static Pod](https://kubernetes.io/docs ...
+ * **Upgrades between minor versions** --- The [`kubeadm upgrade`](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-upgrade/) command is now fully GA. It handles control plane upgrades for you, which includes upgrades to [etcd](https://etcd.io), the [API Server](https://kubernetes.io/docs/reference/using-api/api-overview/), the [Controller Manager](https://kubernetes.io/docs/refer ...
```
Line 50 below, sounds a bit odd:
..organization with special needs?
Possibly change to something like further cluster customization

kbhawkey commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -32,10 +32,10 @@ General Availability means different things for different projects. For kubeadm,
  * **Configuration file schema** --- With the new **v1beta1** API version, you can now tune almost every part of the cluster declaratively and thus build a "GitOps" flow around kubeadm-built clusters. In future versions, we plan to graduate the API to version **v1** with minimal changes (and perhaps none).
- * **The "toolbox" interface of kubeadm** --- Also known as **phases**. If you don't want to consume the all of the `kubeadm init` tasks performed, you can perform fine-grained, generic actions with the `kubeadm init phase` command (for example generating certificates or control plane [Static Pod](https://kubernetes.io/docs/tasks/administer-cluster/static-pod/) manifests).
- * **Upgrades between minor versions** --- The `kubeadm upgrade` command is now fully GA. It handles control plane upgrades for you, which includes upgrades to [etcd](https://etcd.io), the [API Server](https://kubernetes.io/docs/reference/using-api/api-overview/), the [Controller Manager](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/), and the [Schedul ...
+ * **The "toolbox" interface of kubeadm** --- Also known as **phases**. If you don't want to consume the all of the [`kubeadm init`](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/) tasks performed, you can perform fine-grained, generic actions with the `kubeadm init phase` command (for example generating certificates or control plane [Static Pod](https://kubernetes.io/docs ...
```
If you don't want to consume all of the ...

kbhawkey commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -4,7 +4,7 @@ title: Production-Ready Kubernetes Cluster Creation with kubeadm
+**Authors**: Lucas Käldström (CNCF Ambassador) and Luc Perkins (CNCF Developer Advocate)
 
 [kubeadm](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/) is a tool that enables Kubernetes administrators to quickly and easily bootstrap minimum viable clusters that are fully compliant with [Certified Kubernetes](https://github.com/cncf/k8s-conformance/blob/master/terms-conditions/Certified_Kubernetes_Terms.md) guidelines. It's been under active development by [SIG Cluste ...
 
```
does GA imply stable?

neolit123 replied:
yes

kbarnard10 commented:
/hold

benmoss commented on content/en/blog/_posts/2018-12-05-kubeadm-ga-release.md:
```diff
@@ -0,0 +1,97 @@
+We now consider kubeadm to have achieved GA-level maturity in each of these important domains:
+
+ * **Stable command-line UX** --- The kubeadm CLI conforms to [#5a GA rule of the Kubernetes Deprecation Policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli), which states that a command or flag that exists in a GA version must be kept for at least 12 months after deprecation.
+ * **Stable underlying implementation** --- kubeadm now creates a new Kubernetes cluster using methods that shouldn't change any time soon. The control plane, for example, is run as a set of static Pods, bootstrap tokens are used for the [`kubeadm join`](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-join/) flow, and [ComponentConfig](https://github.com/kubernetes/community/blob ...
```
ComponentConfig link should be updated to https://github.com/kubernetes/enhancements/blob/master/keps/sig-cluster-lifecycle/0014-20180707-componentconfig-api-types-to-staging.md

kbarnard10 commented:
/lgtm 
/approve
/hold cancel
